model/task/task.go

```diff
@@ -958,13 +972,23 @@ func (t *Task) MarkDependenciesFinished(ctx context.Context, finished bool) erro
 				DependencyTaskIdKey: t.Id,
 			}},
 		},
-		bson.M{
-			"$set": bson.M{bsonutil.GetDottedKeyName(DependsOnKey, "$[elem]", DependencyFinishedKey): finished},
-		},
-		options.Update().SetArrayFilters(options.ArrayFilters{Filters: []interface{}{
```

Comment:
`ArrayFilters` is not supported on aggregation updates, so I had to replace this with a functionally equivalent `$map`